packages/firestore/src/local/persistence_promise.ts

```diff
@@ -200,10 +200,26 @@ export class PersistencePromise<T> {
 
   static map<R>(all: Iterable<PersistencePromise<R>>): PersistencePromise<R[]> {
     const results: R[] = [];
-    return PersistencePromise.forEach(all, result => {
-      results.push(result);
```

Comment:
Why did this compile before?  Shouldn't `result` be a `PersistencePromise<R>` and therefore be incompatible with `results.push()` which should have expected an `R`?

Reply:
The type signature for `forEach` just listed `(T) => callback`, but it needs to be `(elem:T) => callback`

Reply:
Aha!  Might be time to add 'parameter' to our typedef tslint rules...  https://palantir.github.io/tslint/rules/typedef/